src/main/java/com/chikli/hudson/plugin/naginator/NaginatorListener.java

```diff
@@ -97,8 +106,16 @@ private int calculateRetryCount(@Nonnull Run<?, ?> r) {
         return n;
     }
     
+    /**
+     * @deprecated use {@link NaginatorScheduleAction} to make a build rescheduled.
+     */
+    @Deprecated
     public boolean scheduleMatrixBuild(AbstractBuild<?, ?> build, List<Combination> combinations, int n) {
-        NaginatorMatrixAction nma = new NaginatorMatrixAction();
+        return scheduleMatrixBuild(build, combinations, n, NaginatorListener.calculateRetryCount(build));
+    }
```

Comment:
I think there's no reason to expose this method.
I marked this deprecated and provide no alternate public method.